### ISSUE TITLE
[wav2vec] not reasonable results

### ISSUE THREAD

**Issue by flozi00** [speechbrain/speechbrain#640]
I am using the recipe posted here https://github.com/speechbrain/speechbrain/issues/633#issuecomment-816603992
by @TParcollet 

After two days of training the WER is still around 55-60, while the pretrained wav2vec model has an validation WER of 12% with CTC decoder using huggingface transformers library.
When removing the NN before wav2vec an mismatch occures. Is it possible to replicate the huggingface ctc-decoder architecture and replacing the ctc with attentional RNN ?
The trainingsloss of huggingface and fairseq implementation is similar to the speechbrain loss, so it looks like over fitting

**Comment by TParcollet** [comment]
I am also observing a not that good behavior on CV_fr. Yes it is feasible, but you will have to do it by yourself for now :/ I will try to investigate this this week, it certainly just is a matter of changing few hyperparameters.

**Comment by TParcollet** [comment]
It potentially is a LR scheduling issue, maybe we should force the annealing linearly of exponentially.

**Comment by TParcollet** [comment]
Well, I think it's not a wav2vec2 problem but more a recipe problem. I did a mistake somewhere, dunno where yet. Because even the wav2vec2 model has high WER compared to what I had before (while keeping the same loss). I think I did something wrong when swapping to joint CTC+Att decoding (it was attention only before) let m see.

**Comment by TParcollet** [comment]
Error spotted, I'm sorry. I'll past the recipe here once I have fixed the issue ... Maybe be tmr

**Comment by flozi00** [comment]
Don't know if it helps you, but with char tokenizer instead of unigram or bpe the results are much better too
I did some changes at decoder and tokenizer, furthermore I deleted one layer from vanilla NN and the loss got around 1 after some minutes. Before it took more than a day to get an loss of one

**Comment by TParcollet** [comment]
problems comes from the tokenizer that wasn't taking the right indices ..


```
# defining tokenizer and loading it
    tokenizer = SentencePiece(
        model_dir=hparams["save_folder"],
        vocab_size=hparams["output_neurons"],
        annotation_train=hparams["train_csv"],
        annotation_read="wrd",
        model_type=hparams["token_type"],
        character_coverage=hparams["character_coverage"],
        bos_id=hparams["bos_index"],
        eos_id=hparams["eos_index"]
    )
```

```
# Decoding parameters
# Be sure that the bos and eos index match with the BPEs ones
blank_index: 0
bos_index: 1
eos_index: 2
```

**Comment by flozi00** [comment]
@TParcollet 
Freezing the wave2vec model increases trainingsspeed and quality in my case. WER 11.71 and CER 7.1 after one epoch which took only three hours instead of 12 with unfreezed model.
The loss even is the half after epoch with freezed w2v

**Comment by TParcollet** [comment]
THe training speed is normal: No backward on wav2vec2 with freeze = True. However, the results should be way better when freeze= False.

**Comment by flozi00** [comment]
I did many runs now
The w2v model itself has an WER of around 12%
The model trained with speechbrain is around 11,5% with much more computing power
Double inference time for only an half percent point is not very efficient
Does it match with your experiments @TParcollet ?

**Comment by TParcollet** [comment]
It depends, what are you talking about ? Which dataset, wav2vec2 is frozen or not ? What do you call w2v model itself?

**Comment by flozi00** [comment]
https://huggingface.co/maxidl/wav2vec2-large-xlsr-german
The wave2vec model with linear layer on top for ctc

Commonvoice german, the same one as maxidl used for training
Wav2vec unfrozen has higher WER (13%) as before

**Comment by TParcollet** [comment]
I don't know what to say ... for French the w2v is at 25%. While I'm already at 13% (and still training). However, the fact that frozen works better makes no sense to me :o

**Comment by TParcollet** [comment]
(And the HF model is finetuned also)

**Comment by flozi00** [comment]
Did you changed anything on the recipe ?

**Comment by flozi00** [comment]
I will copy your recipe again, maybe I forgot to undo something after experimenting

**Comment by TParcollet** [comment]
I don't think so ..

**Comment by TParcollet** [comment]
I don't think so ... 
Is it CommonVoice German ? I can give it a try.

**Comment by flozi00** [comment]
I copied the recipe again, WER of 14 after one epoch.
Is it possible to use multiple augmentation methods ?
Either parallel or combined into one ?
For example domaintimeaugment + corrupted env ?

**Comment by TParcollet** [comment]
You can't use specaugment with w2v. I'll try to see.

**Comment by flozi00** [comment]
Not specaugment.
But in real tests only little bit of noise is enough to decrease the accuracy by 10%, the corrupted environment augmentation could be pretty helpful.
Before I was using pydub, audiomentations and so on, but native support would be cleaner solution. 
Do you know this library https://github.com/asteroid-team/torch-audiomentations ?
